src/app/recommendations.ts

```diff
@@ -296,4 +296,187 @@ export const RECOMMENDATIONS: Step[] = [
+    action: "Move any providers that should be available to routed components from the component that defines the `RouterOutlet` to the routed component."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Advanced,
+    step: "18.0.0 Use RedirectCommand or new NavigationBehaviorOptions",
+    action: "When a guard returns a `UrlTree` as a redirect, the redirecting navigation will now use `replaceUrl` if the initial navigation was also using the `replaceUrl` option. If this is not desirable, the redirect can configure new `NavigationBehaviorOptions` by returning a `RedirectCommand` with the desired options instead of `UrlTree`."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Basic,
+    step: "18.0.0 Upgrade TypeScript",
+    action: "Update TypeScript to versions 5.4 or newer."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Advanced,
+    step: "18.0.0: Remove deprecated resource cache providers",
+    action: "Remove dependencies of `RESOURCE_CACHE_PROVIDER` since it's no longer part of the Angular runtime."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Advanced,
+    step: "18.0.0: Update Node.js URL parsing in `ServerPlatformLocation`",
+    action: "In `@angular/platform-server` now `pathname` is always suffixed with `/` and the default ports for http: and https: respectively are 80 and 443."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Medium,
+    step: "18.0.0. Use absolute URLs",
+    action: "Provide an absolute `url` instead of using `useAbsoluteUrl` and `baseUrl` from `PlatformConfig`."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Advanced,
+    step: "18.0.0. Switch from `platformDynamicServer` to `platformServer`.",
+    action: "Replace the usage of `platformDynamicServer` with `platformServer`. Also, add an `import @angular/compiler`."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Medium,
+    step: "18.0.0. Remove `ServerTransferStateModule` from app imports",
+    action: "Remove all imports of `ServerTransferStateModule` from your application. It is no longer needed."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Basic,
+    step: "18.0.0. Update `Route.redirectTo` to accept functions",
+    action: "Update all `redirectTo` uses on your `Route` objects to accept functions. If you were previously using a string, you will need to wrap it in a function."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Basic,
+    step: "18.0.0. Use `afterNextRender` after change detection",
+    action: "Change code which relies on change detection timing to always use `afterNextRender` instead of `setTimeout`."
```

Comment:
Do you know what change this was related to? This one doesn't make sense to me. Maybe it was the change to the timing of zone coalescing? If that's the case, we can probably remove this entirely. We marked it as breaking out of caution but have not seen anything break due to the change so it's probably not even worth mentioning.